packages/task/src/browser/task-service.ts

```diff
@@ -409,6 +410,73 @@ export class TaskService implements TaskConfigurationClient {
     }
 
     async runTask(task: TaskConfiguration, option?: RunTaskOption): Promise<TaskInfo | undefined> {
+        const runningTasksInfo: TaskInfo[] = await this.getRunningTasks();
+
+        // check if the task is active
+        const matchedRunningTaskInfo = runningTasksInfo.find(taskInfo => {
+            const taskConfig = taskInfo.config;
+            return this.taskDefinitionRegistry.compareTasks(taskConfig, task);
+        });
+        if (matchedRunningTaskInfo) { // the task is active
+            const taskName = this.taskNameResolver.resolve(task);
+            const terminalId = matchedRunningTaskInfo.terminalId;
+            if (terminalId) {
+                const terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
+                if (terminal) {
+                    this.shell.activateWidget(terminal.id); // make the terminal visible and assign focus
+                }
+            }
+            this.messageService.info(`The task \'${taskName}\' is already active`, 'Terminate Task', 'Restart Task').then(async actionTask => {
```

Comment:
Should not we use async/await (instead of `.then`) and return a task from `this.restartTask`?